src/monitor.c

```diff
@@ -61,28 +101,42 @@ static int compareVideoModes(const void* fp, const void* sp)
         return fm->width - sm->width;
 
     // Lastly sort on refresh rate
-    return fm->refreshRate - sm->refreshRate;
+    refRateDiff = fm->refreshRate - sm->refreshRate;
+    
+    // Limits the minimum absolute value to 1.0, to prevent different modes from
+    // evaluating to equal, which happens when the difference between the
+    // refresh rates is less than 1.0, and this number is truncated to an integer.
+    return (int) copysign(fmax(fabs(refRateDiff), 1.0), refRateDiff);
 }
 
 // Retrieves the available modes for the specified monitor
 //
 static GLFWbool refreshVideoModes(_GLFWmonitor* monitor)
 {
     int modeCount;
-    GLFWvidmode* modes;
+    _GLFWvideoMode* modes2;
 
-    if (monitor->modes)
+    if (monitor->modes2)
         return GLFW_TRUE;
 
-    modes = _glfw.platform.getVideoModes(monitor, &modeCount);
-    if (!modes)
+    modes2 = _glfw.platform.getVideoModes(monitor, &modeCount);
+    
+    if (!modes2)
         return GLFW_FALSE;
 
-    qsort(modes, modeCount, sizeof(GLFWvidmode), compareVideoModes);
+    qsort(modes2, modeCount, sizeof(_GLFWvideoMode), compareVideoModes);
 
     _glfw_free(monitor->modes);
-    monitor->modes = modes;
-    monitor->modeCount = modeCount;
+    monitor->modes = NULL;
+    
+    monitor->modes2 = modes2;
+    monitor->modesPointers = _glfw_realloc(monitor->modesPointers,
+                                           modeCount * sizeof(GLFWvideoMode*));
+    
+    for (int i = 0; i < modeCount; ++i)
+        monitor->modesPointers[i] = (GLFWvideoMode*) &modes2[i];
+    
+    monitor->modeCount2 = modeCount;
 
     return GLFW_TRUE;
 }
```

Comment:
Is the purpose of this function to let platform implementations set the monitor->modes (now monitor->modes2) to NULL to make GLFW query the video modes again, so that GLFW can be aware of system changes to the supported video modes?  I cannot find that this happens in any of the platform implementations.